Source/Matrix.swift

```diff
@@ -98,13 +98,35 @@ open class Matrix<Element: Value>: MutableQuadraticType, Equatable, CustomString
     public convenience init(_ contents: [[Element]]) {
         let rows = contents.count
         let cols = contents[0].count
+        
+        for element in contents {
+            precondition(element.count == cols)
+        }
 
         self.init(rows: rows, columns: cols)
 
         for (i, row) in contents.enumerated() {
             elements.replaceSubrange(i*cols..<i*cols+min(cols, row.count), with: row)
         }
     }
+    
+    /// Construct a Matrix from an array of rows
+    public convenience init<M: LinearType>(_ contents: [M]) where M.Element == Element {
+        
+        let rows = contents.count
+        let cols = Int(contents[0].count)
+        
+        for element in contents {
+            precondition(element.count == cols)
+        }
+        
+        self.init(rows: rows, columns: cols)
+        
+        for (i, row) in contents.enumerated() {
+            elements.replaceSubrange(i*cols..<i*cols+min(cols, row.count), with: row.map{ $0 })
```

Comment:
Good point. Did think it will only work with a pure [Double] instead of the Elements of the ValueArray directly. Changed it.